ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/git/GitServiceClient.java

```diff
@@ -977,9 +980,29 @@ void log(DevMachine devMachine, ProjectConfigDto project, List<String> fileFilte
      * @param plainText
      *         if <code>true</code> the loq response will be in text format
      */
+    @Deprecated
     Promise<LogResponse> log(DevMachine devMachine, Path project, Path[] fileFilter, boolean plainText);
 
     /**
+     * Get log of commits. The result is the list of {@link Revision}, which is returned by callback in
+     * <code>onSuccess(Revision result)</code>.
```

Comment:
Paraphrase this sentence, this api method returns a promise, it doesn't operate with callbacks.